docs/pages/_app.js

```diff
@@ -60,6 +60,7 @@ ponyfillGlobal.muiDocConfig = {
 
     if (newDeps['@mui/x-date-pickers']) {
       newDeps['@mui/material'] = versions['@mui/material'];
+      newDeps['date-fns'] = 'latest';
```

Comment:
Demos use `data-fns` but it is not installed in the dependencies, so I added `date-fns: 'latest'`

Reply:
I also fixed this in #4508

Reply:
@m4theushw I let you add the second line such that pro demo works too, and I close this PR?

Reply:
I only added to `@mui/x-date-pickers` because even Pro demos need it to import `AdapterDateFns`. This PR is still useful to fix

Reply:
Hello, the issue still persists, I see an error message when I open the demo in codesandbox:
https://codesandbox.io/s/3ruukc?file=/demo.tsx

I added date-fns dependency to my fork, it works now.